### ISSUE TITLE
Loss function usage

### ISSUE THREAD

**Issue by theo1701** [zctao/omnifoldTop#60]
This shouldn't be an urgent problem because things are running fine, but I feel that it is worth looking into.

At model compilation we currently use `loss='binary_crossentropy'`, according to tensor flow documentation here: https://www.tensorflow.org/api_docs/python/tf/keras/Model#compile, it seems like it will use the corresponding loss **function** when a string is given.

If we directly look for the loss function named binary crossentropy, https://www.tensorflow.org/api_docs/python/tf/keras/metrics/binary_crossentropy, it seems like it is not using the `sample_weight`

I'm not entirely sure whether this is related to any issue we have at the moment. I would expect not using sample_weight entirely would lead to more catastrophic problems, rather than just misfits at some regions. But this is definitely worth some testing and clarifications in the future.
